subsys/net/lib/coap/coap_client.c

```diff
@@ -779,6 +781,16 @@ static int handle_response(struct coap_client *client, const struct coap_packet
 		}
 	}
 
+	/* MID-based deduplication */
+	uint16_t response_id = coap_header_get_id(response);
```

Comment:
We usually try to introduce variables at the start of the block

Reply:
That policy isn't even followed by the function I added the variable to :D but sure, fixed